pages/wiki/bitbake.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Working with bitbake
+layout: documentation
+---
+
+
+# Background and Prerequisites
+
+---
+
+The [OpenEmbedded]({{rel 'wiki/openembedded'}}) page describes the overall build system used by AsteroidOS.  It's highly recommended to read that first to get an overview of the build system.  This page is a description of how to use one of the tools, `bitbake` in the context of AsteroidOS.  It is *not* intended to be a general tutorial on `bitbake`.  For more detailed information, see the links in the [OpenEmbedded]({{rel 'wiki/openembedded'}}) page.
+
+The rest of this description assumes that you have already read and followed the [Building AsteroidOS]({{rel 'wiki/building-asteroidos'}}) instructions, and that you have built AsteroidOS for at least one platform (a particular watch or the emulator).
+
+This guide is for those who wish to go deeper and would like to alter or augment the AsteroidOS software.
+
+# Changing Things
+
+---
+
+This will guide you step-by-step from the process of wanting to make a change to actually making that change and getting it running on your watch.
+
+Let's say that you want to make a simple change to the flashlight app on the watch.  Your favorite color is blue, so you want to change the flashlight screen's color to a blue color.  
+
+Since you have already run `bitbake asteroid-image` you know that there are over 5000 tasks to create the image file that can be flashed to a watch.  How do you figure out which of those 5000 tasks builds the flashlight application?
+
+Two fundamental concepts within Yocto and bitbake are the *layer* and the *recipe*.  Layers are the way that, for example, the specific watch hardware can be accomodated and then the asteroid layer can be built on top without changes.  A recipe provides instructions for creating some specific unit of software (a program or library, for instance) just as in cooking, a recipe provides instructions for how to prepare a specific food.  
+
+We can start by examining the layers.  
+
+```
+bitbake-layers show-layers
+NOTE: Starting bitbake server...
+layer                 path                                      priority
+==========================================================================
```

Comment:
Awesome! This shows that we might have an issue with layer priorities?
@FlorentRevest do you know if this is true?

Reply:
I agree these priorities look a bit odd, maybe something to look into :) although if it hasn't caused problems so far it can probably wait a bit